ada/extensions/src/libadalang-helpers.adb

```diff
@@ -0,0 +1,122 @@
+with Ada.Command_Line;
+with Ada.Text_IO; use Ada.Text_IO;
+
+with GNATCOLL.Projects;  use GNATCOLL.Projects;
+with GNATCOLL.Traces;
+with GNATCOLL.VFS;       use GNATCOLL.VFS;
+
+with Libadalang.Project_Provider; use Libadalang.Project_Provider;
+
+package body Libadalang.Helpers is
+
+   function "+" (S : String) return Unbounded_String
+                 renames To_Unbounded_String;
+   function "+" (S : Unbounded_String) return String renames To_String;
+
+   package body App is
+      procedure Run is
+
+         package String_Vectors is new Ada.Containers.Vectors
+           (Positive, Unbounded_String);
+
+         UFP : Unit_Provider_Reference;
+         --  When project file handling is enabled, corresponding unit provider
+
+         Ctx : Analysis_Context;
+
+         Files : String_Vectors.Vector;
+
+         Units : Unit_Vectors.Vector;
+
+      begin
+         --  Setup traces from config file
+         GNATCOLL.Traces.Parse_Config_File;
+
+         if not Args.Parser.Parse then
+            return;
+         end if;
+
+         Ctx := Create_Context
+           (Charset       => +Args.Charset.Get,
+            Unit_Provider => UFP);
+
+         --  Get explicitly passed files if necessary
+         if not Args.Files_From_Project.Get then
```

Comment:
I suggest to terminate with an error if both `--files-from-project` and additional files are passed on the command-line (otherwise one will get silently discarded, error prone for users).

Reply:
That would seem to go in the direction of my previous comment ...

Reply:
As agreed live, I'll remove the option